src/core/common/settings.cpp

```diff
@@ -440,6 +446,76 @@ otError Settings::DeleteDadInfo(void)
 }
 #endif // OPENTHREAD_CONFIG_DUA_ENABLE
 
+#if OPENTHREAD_CONFIG_BORDER_ROUTING_ENABLE
+otError Settings::SaveOmrPrefix(const Ip6::Prefix &aOmrPrefix)
+{
+    otError     error = OT_ERROR_NONE;
+    Ip6::Prefix prevOmrPrefix;
+    uint16_t    length = sizeof(prevOmrPrefix);
+
+    if ((Read(kKeyOmrPrefix, &prevOmrPrefix, length) == OT_ERROR_NONE) && (length == sizeof(prevOmrPrefix)) &&
+        (prevOmrPrefix == aOmrPrefix))
+    {
+        LogPrefix("Re-saved", "OMR prefix", aOmrPrefix);
+        ExitNow();
+    }
+
+    SuccessOrExit(error = Save(kKeyOmrPrefix, &aOmrPrefix, sizeof(aOmrPrefix)));
+    LogPrefix("Saved", "OMR prefix", aOmrPrefix);
+
+exit:
+    LogFailure(error, "saving OMR prefix", false);
+    return error;
+}
+
+otError Settings::ReadOmrPrefix(Ip6::Prefix &aOmrPrefix) const
+{
+    otError  error;
+    uint16_t length = sizeof(aOmrPrefix);
+
+    aOmrPrefix.Clear();
+    SuccessOrExit(error = Read(kKeyOmrPrefix, &aOmrPrefix, length));
```

Comment:
```suggestion
    SuccessOrExit(error = Read(kKeyOmrPrefix, &aOmrPrefix, sizeof(aOmrPrefix)));
```
nit: since length is always `sizeof(aOmrPrefix)`.

Reply:
It requires a non-const reference to output the length it read.